### ISSUE TITLE
How to use multi/exec with celtuce?

### ISSUE THREAD

**Issue by shveik** [lerouxrgd/clj-lettuce#4]
this is more of a question, as I am not able to find documentation on how to wrap commands in a multi/exec block. I am sure i am missing something basic, since i am a Clojure noob.

**Comment by shveik** [comment]
actually, i might be getting somewhere, i need to validate that this works, but wanted to post it in case anyone can validate it: 

```
(deferred/let-flow [^RedisAsyncCommandsImpl cmds (celtuce-manifold/commands-manifold connector)
                      now-seconds   (now-seconds)
                      redis-future  (.multi cmds)
                      items         (redis/zrangebyscore cmds ft 0 now-seconds)
                      removals      (redis/zremrangebyscore cmds ft 0 now-seconds)
                      res           (.exec cmds)
                      ]
    (first [items removals res])))

**Comment by lerouxrgd** [comment]
Commands `WATCH`, `UNWATCH`, `EXEC`, `MULTI` and `DISCARD` should be available for all apis (async, async, reactive), as [mentioned here](https://github.com/lettuce-io/lettuce-core/wiki/Transactions).
I will add them to the `celtuce.commands` namespace so that you won't have to use type hinting/interop

**Comment by allen-servedio** [comment]
@lerouxrgd - I have a question about the code snippet in the second comment above: if connector is created like this:

```
(def connector (delay (conn/redis-server (str "redis://" (:redis-host env) ":" (:redis-port env)) :auto-flush true)))
```

Is that code correct? Meaning, if I understand the Lettuce documentation correctly, doesn't the `connector` in this case have to be an unshared connection from a connection pool? 

Similar to what is shown here: https://github.com/lettuce-io/lettuce-core/wiki/Connection-Pooling#connection-pool-support . I formed this option because of:
* https://github.com/lettuce-io/lettuce-core/wiki/Connection-Pooling#is-connection-pooling-necessary
* https://github.com/lettuce-io/lettuce-core/wiki/Pipelining-and-command-flushing 
> Sharing a connection between threads is possible but keep in mind:
> The longer commands need for processing, the longer other invoker wait for their results
> You should not use transactional commands (MULTI) on shared connection. If you use Redis-blocking commands (e. g. BLPOP) all invocations of the shared connection will be blocked until the blocking command returns which impacts the performance of other threads. Blocking commands can be a reason to use multiple connections.

Or am I reading too much into what they are saying there?

**Comment by lerouxrgd** [comment]
@allen-servedio yes you're right, currently celtuce's `connector` wraps a single connection that will be re-used across threads.
Therefore, as you mentioned, any long time running command (or transaction) will block other threads. 
As they say in the Lettuce doc you pointed, the need for pooling connections should be limited in practice, but it could be possible to add it to celtuce's `connector` with special options or function.
If you're interested in such a feature please make a separate issue and I'll work on it.

**Comment by allen-servedio** [comment]
Thanks, I have created https://github.com/lerouxrgd/celtuce/issues/6. I also found a similar question here: https://github.com/lettuce-io/lettuce-core/issues/480

**Comment by lerouxrgd** [comment]
Note that as mentioned by @mp911de on [stackoverflow](https://stackoverflow.com/a/42091605) there is no out-of-the-box strategy for transactions in the case of `Redis Cluster` so the implementation of `TransactionalCommands`will be provided for `Redis Server` only.

**Comment by lerouxrgd** [comment]
You can now find examples of transaction support in `transactional-commands-test` for both sync/manifold commands on `redis-server` connector.
It will be released in v0.2.1 very soon.
